pkg/security/securitycontextconstraints/provider_test.go

```diff
@@ -993,3 +993,112 @@ func TestValidateAllowedVolumes(t *testing.T) {
 		}
 	}
 }
+
+// TestValidateAllowPrivilegeEscalation will test that when the SecurityContextConstraints
+// AllowPrivilegeEscalation is false we cannot set a container's securityContext
+// to allowPrivilegeEscalation, but when it is true we can.
+func TestValidateAllowPrivilegeEscalation(t *testing.T) {
+	pod := defaultPod()
+	pe := true
+	pod.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = &pe
+
+	// create a SCC that does not allow privilege escalation
+	scc := defaultSCC()
+	dpe := false
+	scc.AllowPrivilegeEscalation = &dpe
+
+	provider, err := NewSimpleProvider(scc)
+	if err != nil {
+		t.Errorf("error creating provider: %v", err.Error())
+	}
+
+	// expect a denial for this SCC and test the error message to ensure it's related to allowPrivilegeEscalation
+	errs := provider.ValidateContainerSecurityContext(pod, &pod.Spec.Containers[0], field.NewPath(""))
+	if len(errs) != 1 {
+		t.Errorf("expected exactly 1 error but got %v", errs)
+	} else {
+		if !strings.Contains(errs.ToAggregate().Error(), "Allowing privilege escalation for containers is not allowed") {
+			t.Errorf("did not find the expected error, received: %v", errs)
+		}
+	}
+
+	// now add allowPrivilegeEscalation to the SecurityContextConstraints
+	scc.AllowPrivilegeEscalation = &pe
+	errs = provider.ValidateContainerSecurityContext(pod, &pod.Spec.Containers[0], field.NewPath(""))
+	if len(errs) != 0 {
+		t.Errorf("directly allowing privilege escalation expected no errors but got %v", errs)
+	}
+}
+
+// TestValidateDefaultAllowPrivilegeEscalation will test that when the SecurityContextConstraints
+// DefaultAllowPrivilegeEscalation is false we cannot set a container's
+// securityContext to allowPrivilegeEscalation but when it is true we can.
+func TestValidateDefaultAllowPrivilegeEscalation(t *testing.T) {
+	yes := true
+	no := false
+
+	pod := defaultPod()
+	pod.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = &yes
+
+	// create a SCC that does not allow privilege escalation
+	scc := defaultSCC()
+	scc.DefaultAllowPrivilegeEscalation = &no
+	scc.AllowPrivilegeEscalation = &no
+
+	provider, err := NewSimpleProvider(scc)
+	if err != nil {
+		t.Errorf("error creating provider: %v", err.Error())
+	}
+
+	// expect a denial for this SCC and test the error message to ensure it's related to allowPrivilegeEscalation
+	errs := provider.ValidateContainerSecurityContext(pod, &pod.Spec.Containers[0], field.NewPath(""))
+	if len(errs) != 1 {
+		t.Errorf("expected exactly 1 error but got %v", errs)
+	} else {
+		if !strings.Contains(errs.ToAggregate().Error(), "Allowing privilege escalation for containers is not allowed") {
+			t.Errorf("did not find the expected error, received: %v", errs)
+		}
+	}
+
+	// now add DefaultAllowPrivilegeEscalation to the SecurityContextConstraints
+	scc.DefaultAllowPrivilegeEscalation = &yes
+	scc.AllowPrivilegeEscalation = &no
```

Comment:
Given https://github.com/openshift/origin/pull/18398/commits/5ca1e181116b82ed63a3bde33d4fb82572649e9b#diff-0d673702b8c4f7d9b631d3bae9cc2d90R136, isn't this an invalid SCC?

Reply:
Good catch!

Reply:
It also exists in the upstream: https://github.com/kubernetes/kubernetes/blob/24ab69d358faeb3452bb76813fddd62afeefacb6/pkg/security/podsecuritypolicy/provider_test.go#L1290-L1293